src/libs/actions/OnyxUpdateManager.ts

```diff
@@ -27,6 +28,134 @@ Onyx.connect({
     callback: (value) => (lastUpdateIDAppliedToClient = value),
 });
 
+let queryPromise: Promise<Response | Response[] | void> | undefined;
+
+type DeferredUpdatesDictionary = Record<number, OnyxUpdatesFromServer>;
+let deferredUpdates: DeferredUpdatesDictionary = {};
+
+// This function will reset the query variables, unpause the SequentialQueue and log an info to the user.
+function finalizeUpdatesAndResumeQueue() {
+    console.debug('[OnyxUpdateManager] Done applying all updates');
+    queryPromise = undefined;
+    deferredUpdates = {};
+    Onyx.set(ONYXKEYS.ONYX_UPDATES_FROM_SERVER, null);
+    SequentialQueue.unpause();
+}
+
+// This function applies a list of updates to Onyx in order and resolves when all updates have been applied
+const applyUpdates = (updates: DeferredUpdatesDictionary) => Promise.all(Object.values(updates).map((update) => OnyxUpdates.apply(update)));
+
+// In order for the deferred updates to be applied correctly in order,
+// we need to check if there are any gaps between deferred updates.
+type DetectGapAndSplitResult = {applicableUpdates: DeferredUpdatesDictionary; updatesAfterGaps: DeferredUpdatesDictionary; latestMissingUpdateID: number | undefined};
+function detectGapsAndSplit(updates: DeferredUpdatesDictionary): DetectGapAndSplitResult {
+    const updateValues = Object.values(updates);
```

Comment:
Do we need to sort updateValues by `previousUpdateID` ? Or they're sorted already?

Reply:
They are sorted by previousUpdateID. When you create an object, it is ordered by the keys - and then when you use `Object.values` it will create an ordered array based on those keys.